packages/node_modules/@ciscospark/widget-recents/src/selector.js

```diff
@@ -126,30 +126,36 @@ const getRecentSpaces = createSelector(
 );
 
 
-const getRecentSpacesWithAvatarUrl = createSelector(
-  [getRecentSpaces, getAvatars],
-  (recentSpaces, avatars) => {
+const getRecentSpacesWithDetail = createSelector(
+  [getRecentSpaces, getAvatars, getCalls],
+  (recentSpaces, avatars, calls) => {
     const avatarItems = avatars.get('items');
-    if (avatarItems.count()) {
-      return recentSpaces.map((space) => {
-        const s = space;
+
+    return recentSpaces.map((space) => {
+      const s = space;
+      if (avatarItems.count()) {
         if (s.type === 'direct') {
           s.avatarUrl = avatarItems.get(s.toPersonId);
         }
         else {
           s.avatarUrl = avatarItems.get(s.id);
         }
-        return s;
-      });
-    }
-    return recentSpaces;
+      }
+      if (calls) {
```

Comment:
@bzang I think you had a better idea on how to handle matching of calls to spaces?

Reply:
Yea, if a `locus` object has a `conversationUrl` you can use that to match the call to a space.

Reply:
Yea, the convo object actually has a `locusUrl` in it. We may need to alter the action helper and the space Record to actually store the value, but that should allow you to directly get the call instead of filtering.

Reply:
Ok, I think we've got it mapped more efficiently now.